lib/screens/widget_screen.dart

```diff
@@ -61,51 +115,233 @@ class DashboardItemScreen extends StatelessWidget {
             value: WidgetProvider(),
           ),
         ],
-        child: StreamBuilder(
-          stream: channel.stream,
-          builder: (context, snapshot) {
-            if (snapshot.error != null) {
-              return ScreenWithAppBar(
-                appBarTitle: 'Widget details',
-                body: WidgetListErrorScreen(),
-              );
-            } else {
-              return Column(
-                children: [
-                  WidgetStatusHeader(
-                    widgetTitle: getWidgetTitle(widget),
-                    status: getWidgetStatus(widget),
-                    lastUpdated: getLastUpdated(widget),
-                  ),
-                  if (renderWidget(widget))
-                    WidgetDetails(
-                      widget: widget,
-                    ),
-                  if (renderWidget(widget))
-                    OpenUrlButton(
-                      widget: widget,
-                    ),
-                ],
-              );
-            }
-          },
-        ),
+        child: PageView.builder(
+        controller: _controller,
+        scrollDirection: Axis.horizontal,
+        itemCount: getWidgetLength(configProvider, dashboardsProvider),
+        onPageChanged: (widgetIndex) {
+          setState(() {
+            setCurrentWidget(configProvider, dashboardsProvider, widgetIndex);
+            pageNumber = widgetIndex;
+          });
+        },
+        itemBuilder: (context, widgetIndex) {
+          DashboardWidget widget = getNextWidget(configProvider, dashboardsProvider, widgetIndex);
+          return Column(
+            children: [
+              WidgetStatusHeader(
+                widgetTitle: getWidgetTitle(widget) != "" && getWidgetTitle(widget) != null
+                    ? getWidgetTitle(widget)
+                    : getWidgetType(widget),
```

Comment:
[LATER] Could we have method `widget.getWidgetTitleOrType()` instead of this?

Reply:
done